File: recursive_max_value.py
def max(arr, max_val=0):
  if arr == []:
    return max_val
  if arr[0] > max_val:
    max_val = arr[0]
  return max(arr[1:], max_val)

print(max([1,3,123,4,12])) # 123

File: recursiveMaxValue.js
function max(arr, maxVal=Number.NEGATIVE_INFINITY) {
  if (!arr.length) return maxVal;
  if (arr[0] > maxVal) maxVal = arr[0];
  return max(arr.slice(1), maxVal);
}

console.log(max([1,3,123,4,12])); // 123
console.log(max([-1,-3,-123,-4,-12])); // -1